### PR TITLE
adds a method to validate a cuid

### PR DESCRIPTION
**Pull request by lferreira**
Hi guys,

This PR adds a validate method for cuid's.

I don't know if is the correct place to put this. I'm available to change if you want.

Thank's

### CODE REVIEW

**Comment by marktani** [comment]
great, I'll have a look tomorrow :)

File: src/test/java/CuidTest.java [hunk 1]
```diff
@@ -13,13 +13,29 @@ public void createCuid() throws Exception {
...
+    }
+
     private boolean hasNoCollisions() {
         int iterations = 4200000;
         Map<String, String> cuids = new HashMap<>();
 
         for (int i = 0; i < iterations; ++i) {
             String cuid = Cuid.createCuid();
-            if (cuids.containsKey(cuid)) {
+            if (cuids.containsKey(cuid) && Cuid.validate(cuid)) {
```
Comment (marktani) [inline src/test/java/CuidTest.java]: I don't think the second condition should be here. What we do with `cuids.containsKey(cuid)` is to check if we already produced this cuid before.

File: src/test/java/CuidTest.java [hunk 2]
```diff
@@ -13,13 +13,29 @@ public void createCuid() throws Exception {
...
+    public void shouldValidateACuid() throws Exception {
+        final String cuid = Cuid.createCuid();
+        assertTrue(Cuid.validate(cuid));
+    }
+
+    @Test
+    public void shouldNotAcceptNullCuidWhenValidate() throws Exception {
+        assertFalse(Cuid.validate(null));
+    }
+
```
Comment (marktani) [inline src/test/java/CuidTest.java:32]: can you add another test case that tests a few strings that are not cuids? Such as 

`hi`, `dh72gsb320000udocl363eofy`, `ch72gsb3200*0udocl363eofy`, `ch72gsb3200p0udocl363eofy`, `ch72gsb3200*0udocl363eof`

File: src/test/java/CuidTest.java (hunk 1 above)
Comment (lferreira) [reply src/test/java/CuidTest.java]: Yeah I agree with you, actually this guy just are there to make sure it's valid in '4200000' iteractions. I'll remove that.

File: src/test/java/CuidTest.java (hunk 2 above)
Comment (lferreira) [reply src/test/java/CuidTest.java:32]: Sure , actually I forgot to validate special chars that we don't have to accept.

I'll do this now

**Comment by marktani** [comment]
thanks for the help @lferreira :slightly_smiling_face:

**Pull request merged**